app/workers/concerns/dump_commons.rb

```diff
@@ -0,0 +1,41 @@
+module DumpCommons
+  def csv_file_path
+    @csv_tempfile ||= Tempfile.new(['export', '.csv'])
+    @csv_tempfile.path
+  end
+
+  def gzip_file_path
+    @gzip_tempfile ||= Tempfile.new(['export', '.gz'])
+    @gzip_tempfile.path
+  end
+
+  def upload_dump
+    to_gzip
+    write_to_s3
```

Comment:
I'd prefer if modules don't depend om methods they don't themselves define (or include automatically).

In fact, I'm thinking maybe this can just be a class that inside `def initialize` makes the csv tempfile, and then has a `to_gzipped` method that opens a gzip tempfile, zips it up and returns the tempfile object. And of course it should have a `close` method that closes both of those.

It could even handle the CSV opening, and just expose methods like "set_headers" and "write_row", and completely encapsulate the CSV generation.